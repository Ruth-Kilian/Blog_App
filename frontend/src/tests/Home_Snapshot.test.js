/* Snapshot test for Home page list */

import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom"; // Import BrowserRouter
import HomePage from "../components/Home/Home";

test("renders HomePage component", () => {
  // mock function for the users
  const setUsers = jest.fn();
  // mock function for the blogs
  const setAllBlogs = jest.fn();
  const allBlogs = [
    // Mock data for testing
    {
      _id: "1",
      title: "Test Blog 1",
      content: "Lorem ipsum dolor sit amet",
      user: {
        _id: "1",
        username: "testuser1",
        profilePicture: "profile1.jpg",
      },
      likesCount: 10,
    },
    {
      _id: "2",
      title: "Test Blog 2",
      content: "Lorem ipsum dolor sit amet",
      user: {
        _id: "2",
        username: "testuser2",
        profilePicture: "profile2.jpg",
      },
      likesCount: 5,
    },
  ];
  const users = [
    // Mock data for testing
    {
      _id: "1",
      username: "testuser1",
      profilePicture: "profile1.jpg",
    },
    {
      _id: "2",
      username: "testuser2",
      profilePicture: "profile2.jpg",
    },
  ];
  // mock function to truncate the content
  const truncateContent = jest.fn();

  // render the component
  const { asFragment } = render(
    // wrap component with Router
    <Router>
      <HomePage
        setUsers={setUsers}
        setAllBlogs={setAllBlogs}
        allBlogs={allBlogs}
        users={users}
        truncateContent={truncateContent}
      />
    </Router>
  );
  // assert the test
  expect(asFragment()).toMatchSnapshot();
});
