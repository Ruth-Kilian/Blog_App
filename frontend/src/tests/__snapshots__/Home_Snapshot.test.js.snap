// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
      style="margin-top: 40px;"
    >
      <div
        class="col-md-6"
        style="margin-bottom: 40px;"
      >
        <a
          class="blog-link-home"
          href="/blog/1"
        >
          <div
            class="mb-3 blog-card-home card"
          >
            <img
              alt="Blog"
              class="card-img-top card-img-top"
              src="http://localhost:8080/uploads/undefined"
            />
            <div
              class="card-body-home card-body"
            >
              <div
                class="card-title-home card-title h5"
              >
                Test Blog 1
              </div>
              <p
                class="card-text-home card-text"
              />
              <div
                class="d-flex justify-content-between align-items-center"
              >
                <div
                  class="picture-home"
                >
                  <img
                    alt="Profile"
                    src="http://localhost:8080/profilePics/profile1.jpg"
                  />
                </div>
                <div
                  class="username-container-home"
                >
                  <p
                    class="mb-0"
                  >
                    <i>
                      testuser1
                    </i>
                  </p>
                </div>
                <p
                  class="mb-0"
                >
                  <i>
                     10 
                    <svg
                      color="red"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: red;"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                      />
                    </svg>
                  </i>
                </p>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        class="col-md-6"
        style="margin-bottom: 40px;"
      >
        <a
          class="blog-link-home"
          href="/blog/2"
        >
          <div
            class="mb-3 blog-card-home card"
          >
            <img
              alt="Blog"
              class="card-img-top card-img-top"
              src="http://localhost:8080/uploads/undefined"
            />
            <div
              class="card-body-home card-body"
            >
              <div
                class="card-title-home card-title h5"
              >
                Test Blog 2
              </div>
              <p
                class="card-text-home card-text"
              />
              <div
                class="d-flex justify-content-between align-items-center"
              >
                <div
                  class="picture-home"
                >
                  <img
                    alt="Profile"
                    src="http://localhost:8080/profilePics/profile2.jpg"
                  />
                </div>
                <div
                  class="username-container-home"
                >
                  <p
                    class="mb-0"
                  >
                    <i>
                      testuser2
                    </i>
                  </p>
                </div>
                <p
                  class="mb-0"
                >
                  <i>
                     5 
                    <svg
                      color="red"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: red;"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                      />
                    </svg>
                  </i>
                </p>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
